src/types.ts

```diff
@@ -1,19 +1,13 @@
-import { IComputedValue, IReactionOptions } from 'mobx';
+import { IReactionOptions } from 'mobx';
 import { StorageAdapter } from './StorageAdapter';
 
-export type PersistenceStore<T> = T & { _asJS: IComputedValue<string>; _storageName: string; _isPersistence: boolean };
-
-export type PersistenceDecoratorOptions = {
+export type PersistenceOptions = {
   name: string;
   properties: string[];
   adapter: StorageAdapter;
   reactionOptions?: IReactionOptions;
 };
 
-export type PersistenceCreatorReturnFunction = <T extends { new (...args: any): {} } | Object>(
-  target: T,
-) => PersistenceStore<T>;
-
 export type StorageAdapterOptions = {
   write: (name: string, value: string) => Promise<Error | void>;
   read: (name: string) => Promise<string | undefined>;
```

Comment:
Looking at `localStorage`, `sessionStorage`, `AsyncStorage` there is a pattern of using `setItem`, `getItem`, and `removeItem` for the storage api. How do you feel about doing the same thing for `StorageAdapterOptions`?

```suggestion
  setItem: (name: string, value: string) => Promise<Error | void>;
  getItem: (name: string) => Promise<string | undefined>;
  removeItem: (name: string) => Promise<void>;
```